specification.markdown

```diff
@@ -749,6 +749,56 @@ type: Unsigned Integer
 
 description: The minimum `DocType` version an `EBML Reader` has to support to read this `EBML Document`. The value of the `DocTypeReadVersion Element` MUST be less than or equal to the value of the `DocTypeVersion Element`.
 
+### DocTypeExtension Element
+
+name: `DocTypeExtension`
+
+path: `0*(\EBML\DocTypeExtension)`
+
+id `0x4281`
+
+minOccurs: 0
+
+type: `Master Element`
+
+description: A `DocTypeExtension` adds extra `Elements` to the main `DocType`+`DocTypeVersion` tuple it's attached to. An `EBML Reader` MAY understand these extra `Elements`. A `DocTypeExtension` MAY be used to iterate between experimental `Elements` before they are integrated in a regular `DocTypeVersion`. Reading one `DocTypeExtension` version of a `DocType`+`DocTypeVersion` tuple doesn't imply one should be able to read upper values of this `DocTypeExtension`.
```

Comment:
I'm not exactly sure what "An `EBML Reader` MAY understand these extra `Elements`." means.

Reply:
I don't think we should allow multiple extension of the same name at all. To do so it may be better to leave it as undefined behaviour so that people don't even try. So we should say the `DocTypeExtensionName` SHOULD be unique in the whole EBML header. But no more.

Reply:
> I'm not exactly sure what "An `EBML Reader` MAY understand these extra `Elements`." means.

It may or may not know what this extension means in terms of element IDs found. In fact it may also alter the interpretation of some other things in the format. For now it's left open.